src/extra-fields/edit.js

```diff
@@ -0,0 +1,246 @@
+			} )
+			.catch( ( err ) => {
+				setError( err.message );
+				setIsLoading( false );
+			} );
+	}, [ userId ] );
+
+	// Apply max fields limit for preview
+	const displayFields = maxFields > 0 ? fields.slice( 0, maxFields ) : fields;
+
+	// Extract background color for cards style
+	const getCardStyle = () => {
+		const isCardsStyle = attributes.className?.includes( 'is-style-cards' );
+		if ( ! isCardsStyle ) {
+			return {};
+		}
+
+		// Get background color from block attributes
+		const style = attributes.style || {};
+		const backgroundColor = attributes.backgroundColor;
+		const customColor = style.color?.background;
+
+		if ( backgroundColor ) {
+			return {
+				backgroundColor: `var(--wp--preset--color--${ backgroundColor })`,
+			};
+		} else if ( customColor ) {
+			return {
+				backgroundColor: customColor,
+			};
+		}
+
+		return {};
+	};
+
+	const cardStyle = getCardStyle();
+
+	const settingsPanel = (
+		<InspectorControls>
+			<PanelBody title={ __( 'Settings', 'activitypub' ) } initialOpen={ true }>
+				<SelectControl
+					label={ __( 'User', 'activitypub' ) }
+					value={ selectedUser }
+					options={ userOptions }
+					onChange={ ( value ) => setAttributes( { selectedUser: value } ) }
+				/>
+				<RangeControl
+					label={ __( 'Maximum Fields', 'activitypub' ) }
+					value={ maxFields }
+					onChange={ ( value ) => setAttributes( { maxFields: value } ) }
+					min={ 0 }
+					max={ 20 }
+					help={ __( 'Limit the number of fields displayed. 0 = show all.', 'activitypub' ) }
+				/>
+			</PanelBody>
+		</InspectorControls>
+	);
+
+	// Render placeholder if inherit mode but no author. Keep controls mounted for recovery.
+	if ( selectedUser === 'inherit' && ! authorId ) {
+		return (
+			<>
+				{ settingsPanel }
+				<div { ...blockProps }>
+					<Placeholder label={ __( 'Fediverse Extra Fields', 'activitypub' ) } icon="list-view">
+						<p>
+							{ __(
+								'This block will display extra fields based on the post author when published.',
+								'activitypub'
+							) }
+						</p>
+					</Placeholder>
+				</div>
+			</>
+		);
+	}
+
+	// Render loading state
+	if ( isLoading ) {
+		return (
+			<div { ...blockProps }>
+				<Placeholder label={ __( 'Fediverse Extra Fields', 'activitypub' ) } icon="list-view">
+					<Spinner />
+				</Placeholder>
+			</div>
+		);
+	}
+
+	// Render error state
+	if ( error ) {
+		return (
+			<div { ...blockProps }>
+				<Placeholder label={ __( 'Fediverse Extra Fields', 'activitypub' ) } icon="list-view">
+					<p>
+						{ __( 'Error loading extra fields:', 'activitypub' ) } { error }
+					</p>
+				</Placeholder>
+			</div>
+		);
+	}
+
+	// Render empty state
+	if ( displayFields.length === 0 ) {
+		return (
+			<>
+				{ settingsPanel }
+				<div { ...blockProps }>
+					<Placeholder label={ __( 'Fediverse Extra Fields', 'activitypub' ) } icon="list-view">
+						<p>{ __( 'No extra fields found. Add fields in your profile settings.', 'activitypub' ) }</p>
```

Comment:
Can we link to the profile settings here? Or directly to the editor to add an extra field?